contrib/platform/src/com/sun/jna/platform/win32/User32Util.java

```diff
@@ -243,6 +247,32 @@ public void run() {
         public void exit() {
             User32.INSTANCE.PostThreadMessage(nativeThreadId, WinUser.WM_QUIT, null, null);
         }
+
+
+        /**
+         * Load a string value from the string table of an executable.
+         * 
+         * @param location the location, eg. %SystemRoot%\system32\input.dll,-5011
+         * @return
+         */
+    	public static String loadString(String location) {
+    		int x = location.lastIndexOf(',');
+    		String moduleName = location.substring(0, x);
+    		int index = Math.abs(Integer.parseInt(location.substring(x + 1)));
+    		String path = Kernel32Util.expandEnvironmentStrings(moduleName);
+    		HMODULE target = Kernel32.INSTANCE.LoadLibraryEx(path, null, Kernel32.LOAD_LIBRARY_AS_DATAFILE);
+    		if (target == null) {
+    			throw new Win32Exception(Kernel32.INSTANCE.GetLastError());
+    		}
+    		PointerByReference lpBuffer = new PointerByReference();
+    		UINT id = new UINT(index);
+    		x = User32.INSTANCE.LoadString(target, id, lpBuffer, 0);
+    		if (0 == x) {
+    			throw new Win32Exception(Kernel32.INSTANCE.GetLastError());
+    		}
+    		return new String(lpBuffer.getValue().getCharArray(0, x));
```

Comment:
This looks suspicious. As written above, you could get a non unicde value, than you are interested in the byte array. And the `Native#toString` functions.

Reply:
Pointer.getString() and Pointer.getWideString() cannot be used as the returned buffer might not be null terminated.
I am now distinguishing between Unicode and non-unicode.